habitat/config/default.py

```diff
@@ -200,6 +200,47 @@
 # for the caveats that results in
 _C.SIMULATOR.HABITAT_SIM_V0.GPU_GPU = False
 # -----------------------------------------------------------------------------
+# PYROBOT
+# -----------------------------------------------------------------------------
+_C.PYROBOT = CN()
+_C.PYROBOT.ROBOTS = ["locobot"]  # types of robots supported
+_C.PYROBOT.ROBOT = "locobot"
+_C.PYROBOT.SENSORS = ["RGB_SENSOR", "DEPTH_SENSOR", "BUMP_SENSOR"]
+_C.PYROBOT.BASE_CONTROLLER = "proportional"
+_C.PYROBOT.BASE_PLANNER = "none"
+# -----------------------------------------------------------------------------
+# SENSORS
+# -----------------------------------------------------------------------------
+PYROBOT_VISUAL_SENSOR = CN()
+PYROBOT_VISUAL_SENSOR.HEIGHT = 480
+PYROBOT_VISUAL_SENSOR.WIDTH = 640
+# -----------------------------------------------------------------------------
+# RGB SENSOR
+# -----------------------------------------------------------------------------
+_C.PYROBOT.RGB_SENSOR = PYROBOT_VISUAL_SENSOR.clone()
+_C.PYROBOT.RGB_SENSOR.TYPE = "PyRobotRGBSensor"
+# -----------------------------------------------------------------------------
+# DEPTH SENSOR
+# -----------------------------------------------------------------------------
+_C.PYROBOT.DEPTH_SENSOR = PYROBOT_VISUAL_SENSOR.clone()
+_C.PYROBOT.DEPTH_SENSOR.TYPE = "PyRobotDepthSensor"
+_C.PYROBOT.DEPTH_SENSOR.MIN_DEPTH = 0.0
+_C.PYROBOT.DEPTH_SENSOR.MAX_DEPTH = 5.0
+_C.PYROBOT.DEPTH_SENSOR.NORMALIZE_DEPTH = True
+# -----------------------------------------------------------------------------
+# BUMP SENSOR
+# -----------------------------------------------------------------------------
+_C.PYROBOT.BUMP_SENSOR = CN()
+_C.PYROBOT.BUMP_SENSOR.TYPE = "PyRobotBumpSensor"
+# -----------------------------------------------------------------------------
+# ACTIONS LOCOBOT
+# -----------------------------------------------------------------------------
+_C.PYROBOT.LOCOBOT = CN()
+_C.PYROBOT.LOCOBOT.ACTIONS = ["BASE_ACTIONS", "CAMERA_ACTIONS"]
```

Comment:
This way of defining actions does not work well with how we have defined `NAVIGATION_ACTIONS` above (line 53). Until now as we have been closely tied to `habitat-sim` most of our API around the simulator has become tightly wound to it. I think being tightly wound on the environment level makes sense here but we should keep things open on the simulator level.

In my integration of PyRobot I have exposed the native actions available on the Robot so that the user has the flexibility to use these as it suits him/her.